Signal/src/ProfileFetcherJob.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+@objc
+class ProfileFetcherJob: NSObject {
```

Comment:
not just retained from objc, but any interaction from objc requires being marked as @objc and descending from NSObject. I believe under the hood a little, this has to do with being compatible with dynamic dispatch.